src/main/java/teammates/logic/core/EmailSenderService.java

```diff
@@ -33,4 +31,8 @@ public void sendEmail(EmailWrapper wrapper) throws EmailSendingException {
     // accounts for the many different Exceptions from different email services
     protected abstract void sendEmailWithService(EmailWrapper wrapper) throws Exception;
 
+    static boolean isErrorStatus(int statusCode) {
```

Comment:
Very strictly speaking, not 2XX is `isNotSuccessfulStatus`, e.g. I won't really call 1XX or 3XX errors.